### PR TITLE
feat: switch doc generation to use helm-docs and add a workflow to auto bump the chart version on pr approval

### PR DESCRIPTION
**Pull request by MarshallAsch**
The inital commits here do not include the new generated version of the chart README to show its generation _after_ the GH approval. 

This workflow will update the chart version automatically _after_ receiving an approval from from one of the maintainers. The version will be selected based on the presence of one of the `chart: patch`, `chart: minor`, or `chart:major` labels on the PR. 

Then the chart documentation will be re generated and committed back to the source branch.

Note: the chart testing linter is expected to fail until _after_ the action is run to bump the chart version

### CODE REVIEW

File: .github/workflows/version.yaml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+      - name: Commit Changes
+        run: |
+          git config user.email '<>'
+          git config user.name github-actions
+          if ! git diff-index --quiet HEAD; then
+            git commit -m "chore: bump $APP chart to v${{ steps.get_version.outputs.new }}" --all
+            git push origin "HEAD:$GITHUB_HEAD_REF"
+          fi
+
```
Comment (cilindrox) [inline .github/workflows/version.yaml:119]:
```suggestion
```

**Review by cilindrox — APPROVED** [review]
LGTM 👍

**Review by cookandy — APPROVED** [review]

File: .github/workflows/version.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
+name: Bump Version and Generate Docs
+on:
+  pull_request_review:
+    types: [submitted]
```
Comment (cilindrox) [inline .github/workflows/version.yaml:4]: maybe filter by PRs targetting `main` or repo's default branch?

Also, dunno how I feel running this on the PR vs on commits to `main` - IMO the latter would prevent any churn on the PR branch and might reduce any risks with commits

**Review by cilindrox — APPROVED** [review]
LGTM 👍

**Pull request merged**